internal/expr/typeinfo.go

```diff
@@ -49,10 +49,22 @@ func (f *structField) outerType() reflect.Type {
 	return f.structType
 }
 
-func (f structField) memberName() string {
+func (f *structField) memberName() string {
 	return f.tag
 }
 
+type sliceType struct {
+	sliceType reflect.Type
+}
+
+func (st *sliceType) outerType() reflect.Type {
+	return st.sliceType
+}
+
+func (st *sliceType) memberName() string {
+	return sliceExtention
```

Comment:
Should we panic here and in the one for map? Should this method be moved to be implemented only for structs and that way we can check in the code that we are in fact dealing with the correct thing?

Reply:
I'm reluctant to have a panic here. There is a place that `memberName()` is used in an error message, and even though I believe it is currently unreachable with slices, I would rather have a weird error message than a panic.